connectors/byoc.py

```diff
@@ -107,31 +106,34 @@ async def save(self, connector):
             doc=document,
         )
 
+    async def heartbeat(self, doc_id):
+        await self.update(doc_id=doc_id, doc={"last_seen": iso_utc()})
+
     async def update_filtering_validation(
         self, connector, validation_result, validation_target=ValidationTarget.ACTIVE
     ):
-        doc_to_update = deepcopy(connector.doc_source)
+        filtering = connector.filtering.to_list()
 
-        for filter_ in doc_to_update.get("filtering", []):
+        for filter_ in filtering:
```

Comment:
Just wonder if know why `_` in `filter_` is needed?

Reply:
If you rename it to `filter`, IDE will complain:
```
Shadows built-in name 'filter' 
```

Reply:
That's weird because I don't see any other `filter` names it shadows.

Reply:
`filter` is a built-in class in Python itself -- so adding `_` is the right pattern